core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -1054,6 +1054,24 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
     schedule.initial.flatMap(loop(None, _))
   }
 
+  /**
+   * Repeats this effect until the result exists in the partial function
+   * codomain.
```

Comment:
I'd probably reword this as "until the partial function succeeds on the result"

Reply:
Yup, done.